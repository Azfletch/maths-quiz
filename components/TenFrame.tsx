import { View } from 'react-native'
import React from 'react'
import { TenFrameProps } from '../types'

const TenFrame = (props: TenFrameProps): JSX.Element => {
  const frames = props?.question?.frames

  if (props.question && props.question.frames) {
    return (
      <View>
        <View style={{ flexDirection: 'row', marginLeft: 'auto', marginRight: 'auto' }}>
          <View style={{ padding: 20, borderColor: 'black', borderWidth: 2 }}>
            <View style={{ alignItems: 'center', height: 25, width: 25, backgroundColor: `${frames[0].counterColor}`, borderRadius: 100 }}>

            </View>
          </View>
          <View style={{ padding: 20, borderColor: 'black', borderWidth: 2 }}>
            <View style={{ alignItems: 'center', height: 25, width: 25, backgroundColor: `${frames[1].counterColor}`, borderRadius: 100 }}>

            </View>
          </View>
          <View style={{ padding: 20, borderColor: 'black', borderWidth: 2 }}>
            <View style={{ alignItems: 'center', height: 25, width: 25, backgroundColor: `${frames[2].counterColor}`, borderRadius: 100 }}>

            </View>
          </View>
          <View style={{ padding: 20, borderColor: 'black', borderWidth: 2 }}>
            <View style={{ alignItems: 'center', height: 25, width: 25, backgroundColor: `${frames[3].counterColor}`, borderRadius: 100 }}>

            </View>
          </View>
          <View style={{ padding: 20, borderColor: 'black', borderWidth: 2 }}>
            <View style={{ alignItems: 'center', height: 25, width: 25, backgroundColor: `${frames[4].counterColor}`, borderRadius: 100 }}>

            </View>
          </View>
        </View>
        <View style={{ flexDirection: 'row', marginLeft: 'auto', marginRight: 'auto' }}>
          <View style={{ padding: 20, borderColor: 'black', borderWidth: 2 }}>
            <View style={{ alignItems: 'center', height: 25, width: 25, backgroundColor: `${frames[5].counterColor}`, borderRadius: 100 }}>

            </View>
          </View>
          <View style={{ padding: 20, borderColor: 'black', borderWidth: 2 }}>
            <View style={{ alignItems: 'center', height: 25, width: 25, backgroundColor: `${frames[6].counterColor}`, borderRadius: 100 }}>

            </View>
          </View>
          <View style={{ padding: 20, borderColor: 'black', borderWidth: 2 }}>
            <View style={{ alignItems: 'center', height: 25, width: 25, backgroundColor: `${frames[7].counterColor}`, borderRadius: 100 }}>

            </View>
          </View>
          <View style={{ padding: 20, borderColor: 'black', borderWidth: 2 }}>
            <View style={{ alignItems: 'center', height: 25, width: 25, backgroundColor: `${frames[8].counterColor}`, borderRadius: 100 }}>

            </View>
          </View>
          <View style={{ padding: 20, borderColor: 'black', borderWidth: 2 }}>
            <View style={{ alignItems: 'center', height: 25, width: 25, backgroundColor: `${frames[9].counterColor}`, borderRadius: 100 }}>

            </View>
          </View>
        </View>
      </View>
    )
  } else {
    return <></>
  }

  
}

export default TenFrame