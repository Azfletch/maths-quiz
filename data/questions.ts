export default [
  {
    question: 'How many blue counters?',
    correctAnswer: 4,
    frames: [
      {
        counterColor: 'red'
      },
      {
        counterColor: 'red'
      },
      {
        counterColor: 'red'
      },
      {
        counterColor: 'white'
      },
      {
        counterColor: 'white'
      },
      {
        counterColor: 'white'
      },
      {
        counterColor: 'blue'
      },
      {
        counterColor: 'blue'
      },
      {
        counterColor: 'blue'
      },
      {
        counterColor: 'blue'
      }
    ]
  },
  {
    question: 'How many green counters?',
    correctAnswer: 6,
    frames: [
      {
        counterColor: 'green'
      },
      {
        counterColor: 'green'
      },
      {
        counterColor: 'red'
      },
      {
        counterColor: 'white'
      },
      {
        counterColor: 'white'
      },
      {
        counterColor: 'green'
      },
      {
        counterColor: 'green'
      },
      {
        counterColor: 'green'
      },
      {
        counterColor: 'green'
      },
      {
        counterColor: 'blue'
      }
    ]
  },
  {
    question: 'How many white counters?',
    correctAnswer: 2,
    frames: [
      {
        counterColor: 'red'
      },
      {
        counterColor: 'white'
      },
      {
        counterColor: 'red'
      },
      {
        counterColor: 'green'
      },
      {
        counterColor: 'blue'
      },
      {
        counterColor: 'red'
      },
      {
        counterColor: 'green'
      },
      {
        counterColor: 'blue'
      },
      {
        counterColor: 'blue'
      },
      {
        counterColor: 'white'
      }
    ]
  },
  {
    question: 'How many red counters?',
    correctAnswer: 8,
    frames: [
      {
        counterColor: 'red'
      },
      {
        counterColor: 'red'
      },
      {
        counterColor: 'red'
      },
      {
        counterColor: 'red'
      },
      {
        counterColor: 'red'
      },
      {
        counterColor: 'red'
      },
      {
        counterColor: 'green'
      },
      {
        counterColor: 'blue'
      },
      {
        counterColor: 'red'
      },
      {
        counterColor: 'red'
      }
    ]
  },
  {
    question: 'How many blue counters?',
    correctAnswer: 6,
    frames: [
      {
        counterColor: 'red'
      },
      {
        counterColor: 'white'
      },
      {
        counterColor: 'blue'
      },
      {
        counterColor: 'green'
      },
      {
        counterColor: 'blue'
      },
      {
        counterColor: 'blue'
      },
      {
        counterColor: 'green'
      },
      {
        counterColor: 'blue'
      },
      {
        counterColor: 'blue'
      },
      {
        counterColor: 'blue'
      }
    ]
  }
]