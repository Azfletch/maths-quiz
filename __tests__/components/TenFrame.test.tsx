import React from 'react'
import { render } from '@testing-library/react-native'

import TenFrame from '../../components/TenFrame'

import questions from '../../data/questions'

test('Tenframe renders correctly', () => {
  // Arrange
  const [question] = questions

  // Act
  const tenFrame = render(<TenFrame question={question} />)

  // Assert
  expect(tenFrame).toMatchSnapshot()
})