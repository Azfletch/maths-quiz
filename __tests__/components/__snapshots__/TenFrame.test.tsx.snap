// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TenFrame component renders correctly 1`] = `
<View>
  <View
    style={
      {
        "flexDirection": "row",
        "marginLeft": "auto",
        "marginRight": "auto",
      }
    }
  >
    <View
      style={
        {
          "borderColor": "black",
          "borderWidth": 2,
          "padding": 20,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "red",
            "borderRadius": 100,
            "height": 25,
            "width": 25,
          }
        }
      />
    </View>
    <View
      style={
        {
          "borderColor": "black",
          "borderWidth": 2,
          "padding": 20,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "red",
            "borderRadius": 100,
            "height": 25,
            "width": 25,
          }
        }
      />
    </View>
    <View
      style={
        {
          "borderColor": "black",
          "borderWidth": 2,
          "padding": 20,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "red",
            "borderRadius": 100,
            "height": 25,
            "width": 25,
          }
        }
      />
    </View>
    <View
      style={
        {
          "borderColor": "black",
          "borderWidth": 2,
          "padding": 20,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 100,
            "height": 25,
            "width": 25,
          }
        }
      />
    </View>
    <View
      style={
        {
          "borderColor": "black",
          "borderWidth": 2,
          "padding": 20,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 100,
            "height": 25,
            "width": 25,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "marginLeft": "auto",
        "marginRight": "auto",
      }
    }
  >
    <View
      style={
        {
          "borderColor": "black",
          "borderWidth": 2,
          "padding": 20,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 100,
            "height": 25,
            "width": 25,
          }
        }
      />
    </View>
    <View
      style={
        {
          "borderColor": "black",
          "borderWidth": 2,
          "padding": 20,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "blue",
            "borderRadius": 100,
            "height": 25,
            "width": 25,
          }
        }
      />
    </View>
    <View
      style={
        {
          "borderColor": "black",
          "borderWidth": 2,
          "padding": 20,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "blue",
            "borderRadius": 100,
            "height": 25,
            "width": 25,
          }
        }
      />
    </View>
    <View
      style={
        {
          "borderColor": "black",
          "borderWidth": 2,
          "padding": 20,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "blue",
            "borderRadius": 100,
            "height": 25,
            "width": 25,
          }
        }
      />
    </View>
    <View
      style={
        {
          "borderColor": "black",
          "borderWidth": 2,
          "padding": 20,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "blue",
            "borderRadius": 100,
            "height": 25,
            "width": 25,
          }
        }
      />
    </View>
  </View>
</View>
`;
